client/src/components/FilesDialog/FilesInput.vue

```diff
@@ -1,55 +1,50 @@
-<template>
-    <BFormInput v-model="localValue" class="directory-form-input" :placeholder="placeholder" @click="selectFile">
-    </BFormInput>
-</template>
-
-<script>
+<script setup lang="ts">
 import { BFormInput } from "bootstrap-vue";
-import { filesDialog } from "utils/data";
-
-export default {
-    components: { BFormInput },
-    props: {
-        value: {
-            type: String,
-        },
-        mode: {
-            type: String,
-            default: "file",
-        },
-        requireWritable: {
-            type: Boolean,
-            default: false,
-        },
-    },
-    data() {
-        return {
-            localValue: this.value,
-        };
-    },
-    computed: {
-        placeholder() {
-            return `Click to select ${this.mode}`;
-        },
-    },
-    watch: {
-        localValue(newValue) {
-            this.$emit("input", newValue);
-        },
-        value(newValue) {
-            this.localValue = newValue;
-        },
+import { computed } from "vue";
+
+import { filesDialog } from "@/utils/data";
+
+interface Props {
+    value: string;
+    mode?: string;
```

Comment:
Could we restrict this further with a string literal union?

Reply:
Done!